python/lsst/jointcal/utils.py

```diff
@@ -52,41 +58,72 @@ def compute_rms(self, data_refs, visit_list, reference):
 
         Return
         ------
-        geom.Angle
+        astropy.Quantity
             New relative RMS of the matched sources.
-        geom.Angle
+        astropy.Quantity
             New absolute RMS of matched sources.
         """
-        visits_per_dataRef = [dataRef.dataId['visit'] for dataRef in data_refs]
 
-        def compute(catalogs):
-            """Compute the relative and absolute RMS."""
+        # DECAM doesn't have "filter" in its registry, so we have to get filter names from VisitInfo.
+        self.filters = [ref.get('calexp').getInfo().getFilter().getName() for ref in data_refs]
```

Comment:
Ugh. I'd much rather use the the structure we've already built to hold metadata (VisitInfo), and not touch the calexp_md (which I'd have thought wouldn't have Filter in it anyway, since it was sanitized by VisitInfo?). I've complained to the Butler people about the fact that we have to do this because we don't have a standardized set of butler keywords (if we did, we could just get filter from the dataRef). The above code shouldn't read the pixels either, since I'm not requesting anything about the pixels, just the metadata (however it is serialized).